src/operators/class.operators.ts

```diff
@@ -0,0 +1,52 @@
+import { Type } from '@nestjs/common';
+
+import { omitOpenAPIProperties, pickOpenAPIProperties } from './openapi.operators';
+import { omitTransformerProperties, pickTransformerProperties } from './transformer.operators';
+import { omitValidatorProperties, pickValidatorProperties } from './validator.operators';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type Obj = {};
+
+/* Create a derived DTO class from the `Base` DTO class.
+ */
+export function derive<T extends Obj>(
+    Base: Type<T>,
+    name: string,
+): Type<T> {
+    // NB: anonymous classes are named via their assigned variable
```

Comment:
This trick was not easy to discover.

Reply:
I wonder if this is documented behavior, or if it's a non-public API-type thing.